qiskit/visualization/matplotlib.py

```diff
@@ -26,6 +26,7 @@
 import numpy as np
 
 try:
+    from matplotlib import get_backend
     from matplotlib import patches
     from matplotlib import pyplot as plt
     HAS_MATPLOTLIB = True
```

Comment:
If we add
```python
INLINE_MPL_BACKEND = get_backend() in ['module://ipykernel.pylab.backend_inline', 'nbAgg']:
```
at this point, we dont have to call `get_backend` so often and we can keep the list of inline backends in a "single" place (as single as `HAS_MATPLOTLIB` setting)

Reply:
That won't work, the backend can be set at runtime and in jupyter it often is; using a magic. Detecting the backend globally at import time will not detect whether someone is running inside a notebook with the inline backends as desired.